### PR TITLE
Make project build with Xcode 12 #2143

### LINKED ISSUE #2143: Make project build with Xcode 12
Follow up to #2142 

Should also fix #2135, which will work for 11.7, but still doesn't with Xcode 12

### LINKED ISSUE #2203: Xcode 12 web3swift pod build errors
Prerequisites: 
Xcode 12.0.1 (12A7300)
macOS: 10.15.7 (19H2)
cocoapods: 1.10.0.rc.1

So I've cloned the repo, stayed in master branch and applied changes to podfile that are presented in pull request by @vladyslav-iosdev (#2189 ). After "make bootstrap" i got the "web3swift" pod of version 0.9.0 and trying to build the AlphaWallet target on my device I got the error "Swift Compiler Error: 'nil' requires a contextual type". Screenshot attached 

Screen Shot 2020-10-12 at 4 48 30 PM

### PR DESCRIPTION
**Pull request by vladyslav-iosdev**
Closes #2143, closes #2203

### CODE REVIEW

**Comment by hboon** [comment]
Careful to not merge this first as it points to WIP repos.

**Comment by hboon** [comment]
@vladyslav-iosdev I'm getting errors with "Xcode 12.0 (12A7209)". Which version are you using?

Screenshot 2020-10-06 at 12 12 04 PM

**Comment by vladyslav-iosdev** [comment]
@hboon i am using Version 12.0.1 (12A7300)

**Comment by hboon** [comment]
Does it build with iPhone X, iOS 12.4 simulator?

**Comment by vladyslav-iosdev** [comment]
> Does it build with iPhone X, iOS 12.4 simulator, if not, can you make it so?

yep, u right for simulator it doesn't work :( i tried with real device while tested it (before to send).

**Review by hboon — CHANGES_REQUESTED** [review]
- [ ] Can you make it build for the simulator too?

- [ ] Is it possible to make it build with both Xcode 11.7 and Xcode 12?

Seems like the only problem building on 11.7 is because of the introduction of `UIUserInterfaceIdiom.mac`

Use a `default:` + //TODO (to remove the default: when we drop 11.7) if you have to :)

- [ ] Are these changes to these needed? Or are the files no longer needed?

* AlphaWallet.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings
* AlphaWallet.xcworkspace/xcshareddata/xcschemes/AlphaWallet - Europe.xcscheme
* AlphaWallet.xcworkspace/xcshareddata/xcschemes/AlphaWallet.xcscheme
* AlphaWallet.xcworkspace/xcshareddata/xcschemes/AlphaWalletTests.xcscheme
* AlphaWallet.xcworkspace/xcshareddata/xcschemes/AlphaWalletUITests.xcscheme

**Comment by EVanzhula** [comment]
How do you guys even compile this? After "make bootstrap" i get the Result pod of version 3.0.0 which doesn't contain the AnyError struct necessary in various places, such as GetIsERC721ForRicketsContractCoordinator.swift.

**Comment by hboon** [comment]
> How do you guys even compile this? After "make bootstrap" i get the Result pod of version 3.0.0 which doesn't contain the AnyError struct necessary in various places, such as GetIsERC721ForRicketsContractCoordinator.swift.

Would you help to create a new issue for this, and mention which version of Xcode and which branch you are building against?

**Comment by vladyslav-iosdev** [comment]
@hboon check this, it works for me. Both for xCode 11 and 12 versions.

**Review by hboon — CHANGES_REQUESTED** [review]
- [x] I pushed a few commits, can you check and squash them into a single commit?
- [ ] Can you make Travis pass? This is before I add commits: https://travis-ci.com/github/AlphaWallet/alpha-wallet-ios/builds/197168016
- [x] I can't run tests in AppCode with Xcode 12, but it still works with 11.7, so we can go with this first
- [x] Does Eureka and BigInt have upstream versions that works with Xcode 12 so we don't have to fork?
- [x] Next is to get these 4 pods merged before merging this PR:
    - [x] pod 'BigInt', :git => 'https://github.com/vladyslav-iosdev/BigInt.git'
    - [x] pod 'Eureka', :git => 'https://github.com/vladyslav-iosdev/Eureka.git', :branch => 'xcode12'
    - [x] pod 'web3swift', :git => 'https://github.com/vladyslav-iosdev/web3swift.git', :commit => 'd146c93b99208b6f3ff1cdbcbd439a7ce6b8ef0a'
    - [x] pod "Macaw", :git => 'https://github.com/vladyslav-iosdev/Macaw.git', :commit => 'bf608a0abfab1fafe68c82bc0c0ab93377da5e53'
- [x] Also, can the change in `ImportWalletViewController.swift` explicitly include `.phone` as a case? Because phone is the most important idiom for us. Search for: "//NOTE: allow to support version xCode 11.7 and xCode 12"
- [ ] Can you make a PR to merge our Macaw fork too? https://github.com/vladyslav-iosdev/Macaw/commits/alphawallet

**Comment by hboon** [comment]
Also, can the change in `ImportWalletViewController.swift` explicitly include `.phone` as a case? Because phone is the most important idiom for us.

Search for: 

> //NOTE: allow to support version xCode 11.7 and xCode 12

**Comment by vladyslav-iosdev** [comment]
> * [ ]  Can you make a PR to merge our Macaw fork too? https://github.com/vladyslav-iosdev/Macaw/commits/alphawallet

it seems like it already exists https://github.com/AlphaWallet/Macaw/pull/4

**Review by hboon — CHANGES_REQUESTED** [review]
Needs the dependent PR to go through and `Podfile` updated first

**Comment by naiba** [comment]
```
[!] Error installing TrustKeystore
[!] Failed to download 'TrustKeystore': [!] /usr/bin/git -C /var/folders/s6/dt3c7zyx0m7dzps3pvf_89880000gn/T/d20201109-72663-ynlfhr checkout --quiet 0aae827ccfbbcb095bc17753dc75b9bb302392aa

fatal: reference is not a tree: 0aae827ccfbbcb095bc17753dc75b9bb302392aa
```

**Comment by naiba** [comment]
> ```
> [!] Error installing TrustKeystore
> [!] Failed to download 'TrustKeystore': [!] /usr/bin/git -C /var/folders/s6/dt3c7zyx0m7dzps3pvf_89880000gn/T/d20201109-72663-ynlfhr checkout --quiet 0aae827ccfbbcb095bc17753dc75b9bb302392aa
> 
> fatal: reference is not a tree: 0aae827ccfbbcb095bc17753dc75b9bb302392aa
> ```

https://github.com/vladyslav-iosdev/latest-keystore-snapshot/commit/0aae827ccfbbcb095bc17753dc75b9bb302392aa

```
set CryptoSwift version to 1.0
-----
This commit does not belong to any branch on this repository.
```

I modified
```pod
pod 'TrustKeystore', :git => 'https://github.com/vladyslav-iosdev/latest-keystore-snapshot.git', :commit => '0aae827ccfbbcb095bc17753dc75b9bb302392aa'
--- to ---
pod 'TrustKeystore', :git => 'https://github.com/vladyslav-iosdev/latest-keystore-snapshot.git', :branch => 'alphawallet'
```
Then it works, Xcode Version 12.1 (12A7403)

**Comment by vladyslav-iosdev** [comment]
Updated PR with setting branch instead of setting commit

**Comment by hboon** [comment]
@vladyslav-iosdev branch isn't good because it's less predictable. Could it be that the commit is still available in your locale cache but not on GitHub?

**Comment by vladyslav-iosdev** [comment]
@hboon can u check it one more time, now should be ok.

**Comment by hboon** [comment]
`pod install` works without error but `Podfile.lock` is modified. Can you try to get it to work with commit hashes instead of branch names?

**Comment by vladyslav-iosdev** [comment]
> * Are these changes to these needed? Or are the files no longer needed?

removed these changes

**Review by hboon — CHANGES_REQUESTED** [review]
> [!] `CryptoSwift` requires CocoaPods version `>= 1.9.1`, which is not satisfied by your current version, `1.8.4`.

I get this line when I do `pod install` with v1.8.4 (which is what the `Podfile.lock` specifies)

Using v1.9.1 works (I guess we have to update to get a newer CryptoSwift version), but modifies `Podfile.lock`. Can you commit that file too?

---

> pod 'TrustKeystore', :git => 'https://github.com/vladyslav-iosdev/latest-keystore-snapshot.git', :commit=>'c0bdc4f6ffc117b103e19d17b83109d4f5a0e764'

Have you created a PR for this ^?

The ones for web3swift and Macaw seems to be ok, but I'll wait till the PR for TrustKeystore is ready too and merge all the PRs for the dependencies at 1 go.

Otherwise seems good, so just these left:

- [ ] Commit `Podfile.lock`
- [ ] Create PR for TrustKeystore
- [ ] Merge all the PRs for the pod dependencies together
- [ ] Fix up Podfile/Podfile.lock in this PR to reflect the dependencies from previous step
- [ ] Merge this PR

**Comment by vladyslav-iosdev** [comment]
> > pod 'TrustKeystore', :git => 'https://github.com/vladyslav-iosdev/latest-keystore-snapshot.git', :commit=>'c0bdc4f6ffc117b103e19d17b83109d4f5a0e764'
> 
> Have you created a PR for this ^?

But this pr is already merged into AlphaWallet branch, i updated pod file with new source

**Comment by vladyslav-iosdev** [comment]
@hboon i updated changes with TrustKeystore link located to AlphaWallet repo, will update rest after PRs will be merged.

**Comment by hboon** [comment]
@vladyslav-iosdev merged the remaining 2 pods we are dependent on. Can you help to update, squash into a single commit, rebase master and force push?

Getting close!

**Comment by hboon** [comment]
Can you make web3swift, Macaw and Eureka point to the specific commit hashes? Otherwise looks good

File: Podfile
Comment (hboon) [inline Podfile:11]: Can you use a commit hash here too?

File: Podfile
Comment (hboon) [inline Podfile:16]: Must have missed this earlier. Can you make it a hash?

**Review by hboon — APPROVED** [review]
Can you merge the PR after the changes?

File: Podfile
Comment (vladyslav-iosdev) [reply Podfile:16]: yep, i missed this, just a sec

**Comment by vladyslav-iosdev** [comment]
done.

**Comment by hboon** [comment]
@vladyslav-iosdev thanks. Go ahead and merge it

**Comment by vladyslav-iosdev** [comment]
> thanks. Go ahead and merge it

u meant that i merge this PR into master, right?

**Comment by hboon** [comment]
Yes. I assume you didn’t change other things so I didn’t check again :)

**Pull request merged**
